src/storage/garbage_collector.cpp

```diff
@@ -50,13 +72,21 @@ uint32_t GarbageCollector::ProcessDeallocateQueue() {
       }
     }
     txns_to_deallocate_ = std::move(requeue);
+
+    // All of the undo records in my buffers were unlinked before the oldest running txn in the system.
+    // We are now safe to deallocate these buffers because no running transaction should hold a reference to them
+    // anymore
+    while (!buffers_to_deallocate_.empty()) {
```

Comment:
Not a problem with your PR, but we will eventually want to rewrite these using the deferred action framework introduced in #358